### PR TITLE
Blog app - controllers specs

### PR DESCRIPTION
**Pull request by r-ahmed2022**
In this PR, I completed the following:-
- Add RSpec as a gem to the project.
- Created Request spec file for all controllers.
- Tested response status was correct, a correct template was rendered, response body included  placeholder text.

### CODE REVIEW

File: spec/controller/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    before do
+      get user_posts_path(user)
+    end
+
+    it 'index for post works!' do
+      expect(response).to have_http_status(:ok)
+      expect(response.body).to include('posts')
+      expect(response).to render_template(:index)
+    end
+  end
```
Comment (codepantha) [inline spec/controller/posts_controller_spec.rb]: - [ ] Please ensure that your controller specs are in the `spec/requests` directory. By placing controller tests in the /requests directory, you can ensure that your tests are testing the entire application stack and not just individual components 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @r-ahmed2022 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added controller specs 📌 
✔️ No linter errors
✔️ Good commit history

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/controller/posts_controller_spec.rb (hunk 1 above)
Comment (r-ahmed2022) [reply spec/controller/posts_controller_spec.rb]: Done, as advised

File: Gemfile.lock [hunk 2]
```diff
@@ -166,6 +166,10 @@ GEM
       activesupport (= 7.0.6)
```
Comment (HAMMAS-SALEEM) [inline Gemfile.lock:166]: **[OPTIONAL]**

- [ ] It would be nice if you make your commit messages capitalized 👍 to make your pull request professional 😊

**Review by HAMMAS-SALEEM — APPROVED** [review]
# Status: Approved :green_circle: :1st_place_medal: :100:
success

### Hey @r-ahmed2022 👋

Great work on implementing the requested changes. Your project is complete. :blush:
There is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following Requirements
✔️ Great work on making all the tests pass means controllers are working as expected 👍

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile.lock (hunk 2 above)
Comment (r-ahmed2022) [reply Gemfile.lock:166]: Noted, Thank you

**Pull request merged**